data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/export/ExportWorker.java

```diff
@@ -0,0 +1,336 @@
+            long totalRecords = 0L;
+            long successRecords = 0L;
+            long failedRecords = 0L;
+
+            // line count regardless the start line number
+            int recordCount = 0;
+            int lastRecordNumberProcessed = 0;
+            final List<String> records = new ArrayList<>();
+            try (MongoCursor<Document> cursor = col.find(query).iterator()) {
+                while (cursor.hasNext()) {
+                    totalRecords += 1;
+                    recordCount += 1;
+                    if (totalRecords <= startLine) {
+                        continue;
+                    }
+
+                    try {
+                        final JsonWriterSettings writerSettings = JsonWriterSettings.builder()
+                                .outputMode(JsonMode.RELAXED)
+                                .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+                                .build();
+                        final String record = cursor.next().toJson(writerSettings);
+                        records.add(record);
+
+                        if ((recordCount - startLine) % DEFAULT_BATCH_SIZE == 0) {
+                            LOG.debug("Write to buffer for line " + (recordCount - DEFAULT_BATCH_SIZE) + " to " + recordCount);
+                            recordBufferWriter.writeToBuffer(createAcknowledgementSet(partition).orElse(null), records);
+                            records.clear();
+                            lastRecordNumberProcessed = recordCount;
+                        }
+
+                        if (System.currentTimeMillis() - lastCheckpointTime > DEFAULT_CHECKPOINT_INTERVAL_MILLS) {
+                            LOG.debug("Perform regular checkpointing for Data File Loader");
+                            // TODO add checkpoint
+                            //checkpointer.checkpoint(lastLineProcessed);
+                            lastCheckpointTime = System.currentTimeMillis();
+
+                        }
+
+                        successItemsCounter.increment();
+                        successRecords += 1;
+                    } catch (Exception e) {
+                        LOG.error("failed to add record to buffer with error {}", e.getMessage());
+                        failureItemsCounter.increment();
+                        failedRecords += 1;
+                    }
+                }
+
+                final Optional<DataQueryProgressState> progressState = partition.getProgressState();
+                progressState.get().setLoadedRecords(totalRecords);
+                // TODO update progress state
+            } catch (Exception e) {
+                LOG.error("Exception connecting to cluster {}", e.getMessage());
+                throw new RuntimeException(e);
+            }
+
+            LOG.info("Records processed: " + totalRecords + " , recordCount: " + recordCount);
```

Comment:
Please use SLF4J string interpolation instead of string concatenation.

Reply:
Done